controllers/tab/workflow/VersioningTabHandler.inc.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file controllers/tab/workflow/VersioningTabHandler.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class VersioningTabHandler
+ * @ingroup controllers_tab_workflow
+ *
+ * @brief Handle AJAX operations for version tabs on production stages workflow pages.
+ */
+
+import('classes.handler.Handler');
+
+// Import the base class.
+import('lib.pkp.classes.controllers.tab.workflow.PKPVersioningTabHandler');
+
+class VersioningTabHandler extends PKPVersioningTabHandler {
+
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(
+			array(ROLE_ID_SUB_EDITOR, ROLE_ID_MANAGER, ROLE_ID_ASSISTANT),
+			array('versioning', 'newVersion')
+		);
+	}
+
+
+	//
+	// Extended methods from Handler
+	//
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		$stageId = (int) $request->getUserVar('stageId'); // This is validated in WorkflowStageAccessPolicy.
+
+		import('lib.pkp.classes.security.authorization.WorkflowStageAccessPolicy');
+		$this->addPolicy(new WorkflowStageAccessPolicy($request, $args, $roleAssignments, 'submissionId', $stageId));
```

Comment:
I think, every role that has access should be able to create new versions. What do you think? Do I need to change something here to accomplish that?